assets/js/components/Dashboard/NewPaMessage/StationsAndZones/PlaceZonesRow.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { Place } from "Models/place";
+import { Screen } from "Models/screen";
+import { signIDs } from "../../../../util";
+import fp from "lodash/fp";
+import { Button } from "react-bootstrap";
+import SignButtonGroup from "./SignButtonGroup";
+import { ArrowLeftShort, ArrowRightShort } from "react-bootstrap-icons";
+
+interface PlaceZonesRow {
+  place: Place;
+  allSelectedSigns: string[];
+  allSignsForRouteAtPlace: Screen[];
+  setSignIds: (signIds: string[]) => void;
+  leftZones: Screen[];
+  middleZones: Screen[];
+  rightZones: Screen[];
+  route: string;
+  branches?: string[];
+}
+
+const PlaceZonesRow = ({
+  place,
+  allSelectedSigns,
+  allSignsForRouteAtPlace,
+  setSignIds,
+  leftZones,
+  middleZones,
+  rightZones,
+  route,
+  branches = [],
+}: PlaceZonesRow) => {
+  if (allSignsForRouteAtPlace.length === 0) return null;
+
+  const selectedSignsAtPlace = signIDs(allSignsForRouteAtPlace).filter(
+    (id: string) => allSelectedSigns.includes(id),
+  );
+
+  const onSignButtonClick = (signID: string) => {
+    if (allSelectedSigns.includes(signID)) {
+      setSignIds(fp.without([signID], allSelectedSigns));
+    } else {
+      setSignIds(fp.uniq([...allSelectedSigns, signID]));
+    }
+  };
+
+  const allSignsSelected =
+    selectedSignsAtPlace.length === allSignsForRouteAtPlace.length;
```

Comment:
**question:** is this always true?

Reply:
No, `allSignsForRouteAtPlace` contains selected and unselected signs that match the current route filter. `selectedSignsAtPlace` is all of the selected signs from `allSignsForRouteAtPlace`. So when a sign is unselected, it becomes `false`.

Reply:
oh sorry my question was unclear. i meant it is always true that if these have the same length all signs are selected?

Reply:
It is, but I can see how the variable implies something different. `selectedSignsAtPlace` is a bit misleading because it's only the selected signs for the filtered route. I can change that.